pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,460 @@
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPv4 enabled status")
+	}
+	isIPv6Enabled, err := config.IsIPv6Enabled(nodeConfig, trafficEncapMode)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPv6 enabled status")
+	}
+	m.isIPv4Enabled = isIPv4Enabled
+	m.isIPv6Enabled = isIPv6Enabled
+
+	// Add Node informer event handler for Node
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeAdd,
+		UpdateFunc: m.onNodeUpdate,
+		DeleteFunc: m.onNodeDelete,
+	})
+
+	// Add crd informer event handler for NodeLatencyMonitor
+	nlmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeLatencyMonitorAdd,
+		UpdateFunc: m.onNodeLatencyMonitorUpdate,
+		DeleteFunc: m.onNodeLatencyMonitorDelete,
+	})
+
+	return m
+}
+
+// onNodeAdd is the event handler for adding Node.
+func (m *NodeLatencyMonitor) onNodeAdd(obj interface{}) {
+	node := obj.(*corev1.Node)
+	m.latencyStore.addNode(node)
+
+	klog.V(4).InfoS("Node added", "Node", klog.KObj(node))
+}
+
+// onNodeUpdate is the event handler for updating Node.
+func (m *NodeLatencyMonitor) onNodeUpdate(oldObj, newObj interface{}) {
+	oldNode := oldObj.(*corev1.Node)
+	node := newObj.(*corev1.Node)
+	m.latencyStore.updateNode(oldNode, node)
+
+	klog.V(4).InfoS("Node updated", "Node", klog.KObj(node))
+}
+
+// onNodeDelete is the event handler for deleting Node.
+func (m *NodeLatencyMonitor) onNodeDelete(obj interface{}) {
+	// Check if the object is a not a Node
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		// Check if the object is a DeletedFinalStateUnknown in k8s
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(nil, "Received unexpected object", "obj", obj)
+			return
+		}
+		// Convert the DeletedFinalStateUnknown to a Node
+		node, ok = deletedState.Obj.(*corev1.Node)
+		if !ok {
+			klog.ErrorS(nil, "DeletedFinalStateUnknown contains non-Node object", "obj", deletedState.Obj)
+			return
+		}
+	}
+
+	m.latencyStore.deleteNode(node)
+}
+
+// onNodeLatencyMonitorAdd is the event handler for adding NodeLatencyMonitor.
+func (m *NodeLatencyMonitor) onNodeLatencyMonitorAdd(obj interface{}) {
+	nlm := obj.(*v1alpha1.NodeLatencyMonitor)
+	klog.V(4).InfoS("NodeLatencyMonitor added", "NodeLatencyMonitor", klog.KObj(nlm))
+
+	if err := m.updateLatencyConfig(nlm); err != nil {
+		klog.ErrorS(err, "Failed to update latency config")
+	}
+}
+
+// onNodeLatencyMonitorUpdate is the event handler for updating NodeLatencyMonitor.
+func (m *NodeLatencyMonitor) onNodeLatencyMonitorUpdate(oldObj, newObj interface{}) {
+	oldNLM := oldObj.(*v1alpha1.NodeLatencyMonitor)
+	newNLM := newObj.(*v1alpha1.NodeLatencyMonitor)
+	klog.V(4).InfoS("NodeLatencyMonitor updated", "NodeLatencyMonitor", klog.KObj(newNLM))
+
+	if oldNLM.GetGeneration() == newNLM.GetGeneration() {
+		return
+	}
+
+	if err := m.updateLatencyConfig(newNLM); err != nil {
```

Comment:
Let's not add error handling for a function that we write and will never fail apparently

Reply:
OK, I have removed them.